### ISSUE TITLE
expanding drSubset output options

### ISSUE THREAD

**Issue by amwhite** [tesseradata/datadr#56]
Currently drSubset only allows the result to be returned in memory. It would be helpful if it had the option to save the result using and hdfsConn or localDiskConn (similar to the output parameter in functions such as drLapply, drSample, drFilter, etc).

**Comment by hafen** [comment]
Yes - I suppose the question is how to partition the results.  The purpose of `drSubset` was to subset the data to a single data frame, in which case why not pull it into memory.  If you want to subset the data and maintain the same subsetting, you can simply do it with `drLapply`.  If you want a new partitioning you can do it with `addTransform` and `divide`.  What do you think?
